### PR TITLE
[Doc] Tizen Native App Guide

### PR DESCRIPTION
**Pull request by myungjoo**
This guide allows developers to start writing Tizen native apps more easily.

This partially resovles #1751.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: Documentation/writing-tizen-native-apps.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+3. Invoke (provide an input, get an output)
+
+4. Close the handle
+
+Note that set_timeout API allows you to set timeout for invoke function.
+
+## General Flow / Pipeline
+
+It is far more rich featured and provides a lot more functionalities. However, the developer is required to understand GStreamer pipelines. For GStreamer pipelines in general, plrease refer to GStreamer documents.
```
Comment (jaeyun-jung) [inline Documentation/writing-tizen-native-apps.md]: plrease typo

File: Documentation/writing-tizen-native-apps.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+1. Construct a pipeline (ml\_pipeline\_construct) and get a pipeline handle.
+
+2. (Optional) Attach callbacks or get other handles from the constructed pipeline. In order to attach callbacks or get handles, the corresponding elements should have names defined so that they can be recognized in run-time.
+
+3. Start the pipeline. The callbacks will be invoked during the pipeline execution. You may control it with the additional handles (e.g., input/output switches, valves).
+
+4. Pause/Resume the pipeline.
```
Comment (jaeyun-jung) [inline Documentation/writing-tizen-native-apps.md]: we dont have APIs to directly access the element to pause/resume (stop/start only)

File: Documentation/writing-tizen-native-apps.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+5. Stop the pipeline.
+
+6. Close the pipeline handle.
+
+### Sink Callbacks (ml\_pipeline\_sink\_cb)
+
+This allows you to provide a function (callback), which is invoked whenever an output tensor/tensors is available. This can be attached to ```tensor_sink``` with the given name.
```
Comment (jaeyun-jung) [inline Documentation/writing-tizen-native-apps.md]: both tensor_sink and appsink are available

File: Documentation/writing-tizen-native-apps.md (hunk 1 above)
Comment (myungjoo) [reply Documentation/writing-tizen-native-apps.md]: fixed

File: Documentation/writing-tizen-native-apps.md (hunk 2 above)
Comment (myungjoo) [reply Documentation/writing-tizen-native-apps.md]: Fixed.

File: Documentation/writing-tizen-native-apps.md (hunk 3 above)
Comment (myungjoo) [reply Documentation/writing-tizen-native-apps.md]: fixed

File: Documentation/writing-tizen-native-apps.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# Writing a Tizen Native App
+
+This document guides how to write a Tizen Native (C/C++) app with nnstreamer APIs (Tizen Machine-Learning Inference API Set).
+
+## About the Tizen ML-Inference API Sets
+
+There are two API sets as of Tizen 5.5 M2, Machine-Learning Inference: "Pipeline" and "Single".
+
+With **Pipeline** API set, you can construct a GStreamer pipeline with nnstreamer elements and gstreamer elements in the "whitelist". GStreamer plugins not listed in the "whitelist" is not supported by Pipelie APIs. Note that an alias element, ```tizencamvideosrc``` (or ML_TIZEN_CAM_VIDEO_SRC macro), denotes for a video camera source, which may vary per device; in other words, you don't need to worry whether it's ```v4l2src``` or ```camsrc``` or whatsoever. For the list of whitelist, refer to the Tizen API doc or ```/etc/nnstreamer.ini```.
```
Comment (kparichay) [inline Documentation/writing-tizen-native-apps.md]: Changes marked as bold in the paragraph

With Pipeline API set, you can construct a GStreamer pipeline with nnstreamer elements and gstreamer elements in the "whitelist". GStreamer plugins not listed in the "whitelist" **~is~ -> are** not supported by Pipelie APIs. Note that an alias element, ```tizencamvideosrc``` (or ML_TIZEN_CAM_VIDEO_SRC macro), denotes **~for~** a video camera source, which may vary per device; in other words, you don't need to worry whether it's ```v4l2src``` or ```camsrc``` or whatsoever. For the list of whitelist **elements**, refer to the Tizen API doc or ```/etc/nnstreamer.ini```.

File: Documentation/writing-tizen-native-apps.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+3. Invoke (provide an input, get an output)
+
+4. Close the handle
+
+Note that set_timeout API allows you to set timeout for invoke function.
+
+## General Flow / Pipeline
+
+It is far more rich featured and provides a lot more functionalities. However, the developer is required to understand GStreamer pipelines. For GStreamer pipelines in general, please refer to GStreamer documents.
```
Comment (kparichay) [inline Documentation/writing-tizen-native-apps.md]: It is far more rich featured and provides a lot more functionalities **compared to Single API**

File: Documentation/writing-tizen-native-apps.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+3. Release the handle.
+
+### Utility Functions
+
+You may investigate the data types/dimensions/names or get performance-related data.
+
+
+### Element Whitelist
+
+The elements that can be included ihe pipeline (with construct API) are limited by the whitelist (defined by ```/etc/nnstreamer.ini```).
```
Comment (kparichay) [inline Documentation/writing-tizen-native-apps.md]: ihe typo

File: Documentation/writing-tizen-native-apps.md (hunk 5 above)
Comment (myungjoo) [reply Documentation/writing-tizen-native-apps.md]: Refactored.

File: Documentation/writing-tizen-native-apps.md (hunk 4 above)
Comment (myungjoo) [reply Documentation/writing-tizen-native-apps.md]: Fixed.

File: Documentation/writing-tizen-native-apps.md (hunk 6 above)
Comment (myungjoo) [reply Documentation/writing-tizen-native-apps.md]: Fixed.

**Comment by myungjoo** [comment]
@jaeyun-jung @again4you WIth this PR merged, let's try Tizen-next.

**Pull request merged**